tests/test_kernel_pcovr.py

```diff
@@ -177,6 +192,118 @@ def test_centerer(self):
+                str(cm.message),
+                "Kernel parameter mismatch: the regressor has kernel parameters "
+                "{kernel: linear, gamma: 0.2, degree: 3, coef0: 1, kernel_params: None}"
+                " and KernelPCovR was initialized with kernel parameters "
+                "{kernel: linear, gamma: 0.1, degree: 3, coef0: 1, kernel_params: None}",
+            )
+
+    def test_incompatible_regressor(self):
+        regressor = Ridge(alpha=1e-8)
+        regressor.fit(self.X, self.Y)
+        kpcovr = self.model(mixing=0.5, regressor=regressor)
+
+        with self.assertRaises(ValueError) as cm:
+            kpcovr.fit(self.X, self.Y)
+            self.assertTrue(
+                str(cm.message),
+                "Regressor must be an instance of `KernelRidge`",
+            )
+
+    def test_none_regressor(self):
+        kpcovr = KernelPCovR(mixing=0.5, regressor=None)
+        kpcovr.fit(self.X, self.Y)
+        self.assertTrue(kpcovr.regressor is None)
+        self.assertTrue(kpcovr.regressor_ is not None)
+
+    def test_incompatible_coef_shape(self):
+
+        # 1D properties (self.Y is 2D with two targets)
+        # X shape doesn't match
+        regressor = KernelRidge(alpha=1e-8, kernel="linear")
+        regressor.fit(self.X, self.Y[:, 0])
+        kpcovr = self.model(mixing=0.5, regressor=regressor)
+
+        with self.assertRaises(ValueError) as cm:
+            kpcovr.fit(self.X[0:-1], self.Y[0:-1, 0])
+            self.assertTrue(
+                str(cm.message),
+                "The target regressor has a shape incompatible "
+                "with the supplied sample space",
+            )
+
+        # >= 2D properties
+        # Y shape doesn't match
+        regressor = KernelRidge(alpha=1e-8, kernel="linear")
+        regressor.fit(self.X, self.Y[:, 0][:, np.newaxis])
+        kpcovr = self.model(mixing=0.5, regressor=regressor)
+
+        with self.assertRaises(ValueError) as cm:
+            kpcovr.fit(self.X, self.Y[:, 0])
+            self.assertTrue(
+                str(cm.message),
+                "The target regressor has a shape incompatible "
+                "with the supplied target space",
+            )
+
+        with self.assertRaises(ValueError) as cm:
+            kpcovr.fit(self.X, self.Y)
+            self.assertTrue(
+                str(cm.message),
+                "The target regressor has a shape incompatible "
+                "with the supplied target space",
+            )
+
+        # X shape doesn't match
+        regressor = KernelRidge(alpha=1e-8, kernel="linear")
+        regressor.fit(self.X, self.Y)
+        kpcovr = self.model(mixing=0.5, regressor=regressor)
+
+        with self.assertRaises(ValueError) as cm:
+            kpcovr.fit(self.X[0:-1], self.Y[0:-1])
+            self.assertTrue(
+                str(cm.message),
+                "The target regressor has a shape incompatible "
+                "with the supplied sample space",
+            )
+
```

Comment:
So much of this would be avoided by the suggestion above. All you'd have to do is check the regressor output shape